### PR TITLE
BDD extension for cypress testing

### PR DESCRIPTION
**Pull request by ScriptingShrimp**
This PR extends [Initial cypress integration](https://github.com/kiali/kiali-ui/pull/2308) with BDD plugin and simple `.feature` file for learning purposes. 

As QE's, we will extend this, refactor and make comprehensive structure over time. Right now, our QE's are familiar how to run BDD files and how to make small, incremental improvements. 

Already we spent ~4 hours learning and exercising on this approach and so far I have positive feedback.  This is also part of JIRA Ticket 

- to implement [Cypress plugin - Cucumber / BDD](https://issues.redhat.com/browse/OSSM-1039)
- have [Cypress PoC with working login](https://issues.redhat.com/browse/OSSM-1017)

image

### CODE REVIEW

File: cypress-bdd.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "baseUrl": "https://kiali-istio-system.apps.pmdown4.0216-eg3.fw.rhcloud.com",
```
Comment (nrfox) [inline cypress-bdd.json:2]: Can you change this to something else by default? Something like `http://localhost:3000`? The base url will likely need to be updated for whoever is running the tests through the cypress env var or through a command line flag.

File: cypress/integration/common/kiali_homepage.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Given } from "cypress-cucumber-preprocessor/steps";
+import { Then } from "cypress-cucumber-preprocessor/steps";
+
+const url = "https://kiali-istio-system.apps.pmdown4.0216-eg3.fw.rhcloud.com"
```
Comment (nrfox) [inline cypress/integration/common/kiali_homepage.ts:4]: If you set `baseUrl` through an env var you shouldn't need to set this explicitly: https://docs.cypress.io/guides/references/best-practices#Setting-a-global-baseUrl. `cy.visit(url)` then becomes `cy.visit("/")`.

File: cypress/integration/check_graph.spec.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { Given } from "cypress-cucumber-preprocessor/steps";
+
+const url = 'https://google.com'
+Given('user is logged in as administrator', () => {
+  cy.visit(url)
+})
```
Comment (nrfox) [inline cypress/integration/check_graph.spec.ts:6]: This looks like just an example. Can we remove it?

**Review by nrfox — CHANGES_REQUESTED** [review]
Just a couple comments about the hardcoded urls otherwise LGTM

File: cypress-bdd.json (hunk 1 above)
Comment (jshaughn) [reply cypress-bdd.json:2]: +1 to the suggested URL, it works well when already using `yarn start`.

File: cypress/integration/featureFiles/kiali_homepage.feature [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Feature: Kilai login
```
Comment (jshaughn) [inline cypress/integration/featureFiles/kiali_homepage.feature:1]:
```suggestion
Feature: Kiali login
```

File: cypress-bdd.json (hunk 1 above)
Comment (ScriptingShrimp) [reply cypress-bdd.json]: changed to suggested value

**Review by nrfox — APPROVED** [review]
Works great! Excited to begin using this.

**Pull request merged**